src/useCounter/useCounter.ts

```diff
@@ -0,0 +1,116 @@
+import { SetStateAction, useCallback, useRef } from 'react';
+import { IInitialState, resolveHookState } from '../util/resolveHookState';
+import { useSafeState, useSyncedRef } from '..';
+
+export interface CounterActions {
+  /**
+   * Returns the current value of the counter.
+   */
+  get: () => number;
+  /**
+   * Increment the counter by the given `delta`.
+   *
+   * @param `delta` number or function returning a number. By default, `delta` is 1.
+   */
+  inc: (delta?: SetStateAction<number>) => void;
+  /**
+   * Decrement the counter by the given `delta`.
+   *
+   * @param `delta` number or function returning a number. By default, `delta` is 1.
+   */
+  dec: (delta?: SetStateAction<number>) => void;
+  /**
+   * Set the counter to any value, limited only by the `min` and `max` parameters of the hook.
+   *
+   * @param `value` number or function returning a number
+   */
+  set: (value: SetStateAction<number>) => void;
+  /**
+   * Resets the counter to its original initial value.
+   *
+   * If `value` is given, then it becomes the new initial value of the hook and
+   * following calls to `reset` without arguments will reset the counter to `value`.
+   *
+   * @param `value` number or function returning a number
+   */
+  reset: (value?: SetStateAction<number>) => void;
+}
+
+/**
+ * Tracks a numeric value.
+ *
+ * @param initialValue The initial value of the counter.
+ * @param max The maximum value the counter is allowed to reach.
+ *            If `initialValue` is greater than `max`, then `max` is set as the initial value.
+ * @param min The minimum value the counter is allowed to reach.
+ *            If `initialValue` is smaller than `min`, then `min` is set as the initial value.
+ */
+export function useCounter(
+  initialValue: IInitialState<number> = 0,
+  max?: number,
+  min?: number
+): [number, CounterActions] {
+  const resolvedInitialValue = resolveHookState(initialValue);
+
+  const fitInRange = useCallback(
+    (value: number) => {
+      let fittedValue = value;
+
+      if (typeof max !== 'undefined') {
+        fittedValue = Math.min(max, fittedValue);
+      }
+
+      if (typeof min !== 'undefined') {
+        fittedValue = Math.max(min, fittedValue);
+      }
+
+      return fittedValue;
+    },
+    [max, min]
+  );
```

Comment:
It could've been done via `useMediatedState`. Also there would be no need in `useCallback`, as `useMediatedState` perfectly handles dynamic mediators.

Reply:
Interesting! Should get more experience using this library in order to be able to think more through all these hooks.

I see you already committed a change where you implemented this. Is there anything I should still do?

Reply:
No, i'll merge today.